### DISCUSSION TITLE
Query Inteceptor Ability

### DISCUSSION THREAD

**Discussion by wxjlibra** [nebula-contrib/ngbatis#292] [Q&A]
Do we have native support for QueryInteceptor like mybatis org.apache.ibatis.plugin.Interceptor where we could get metadata like query template, and parameters, queryStatementId etc.
Or we need to do ourselves, in that case can suggest.

**Comment by CorvusYe** [comment]
At present, there is no query interception function.

Because the semantic recognition of ngql cannot be achieved, it is difficult to ensure the legitimacy of the intercepted and changed ngql from the framework layer, which may make the intercepted function unstable or make the calling method very rigid. Therefore, this part of the work has not been carried out.

I am currently working on ngql semantic parsing, and if there is a breakthrough, it is possible to implement this function in the future

Perhaps you can describe your specific needs and see if there are other ways to temporarily implement them.
